### PR TITLE
Alternative transmit data: #4351

### PR DESCRIPTION
**Pull request by wimalopaan**
WIP: first try for: https://github.com/EdgeTX/edgetx/discussions/4351

### CODE REVIEW

**Comment by pfeerick** [comment]
Please rebase rather than merge main where possible, as this keeps the commit history and changes clearer.

**Comment by pfeerick** [comment]
If the main branch in your repo was up to date at the time, it should have left you with only the two unique commits, rather than also including these ones. 

image

**Comment by pfeerick** [comment]
If you have no objections, I can push that through as I rebased it locally and just need to push it back to this branch. You'll just need to reset your local copy to the state of the remote branch in order to continue working on it.

**Comment by wimalopaan** [comment]
> If you have no objections, I can push that through as I rebased it locally and just need to push it back to this branch. You'll just need to reset your local copy to the state of the remote branch in order to continue working on it.

Well, I'm not a real git guru ;-) 

We can do as you suggested above (then you have to guide me a bit for my local repo), or I can delete the PR and setup a new branch and PR ... choose the way that means the least work for you!

**Comment by 3djc** [comment]
I think the option should be off by default,  not on

**Comment by pfeerick** [comment]
Yes, any supporting Lua code would go in the edgetx-sdcard repo. 

rebase has been pushed. And don't worry, I'm not a git guru either... I've just learnt a bit over the past few years having to use it more and more ;)

Using the command line, do a `git fetch` in your local working copy, so that git recognises there are changes in the remote. When you do `git status`, it should complain about it being not up to date. Using the branch name as shown here (I'm guessing it will say origin/wmbindata for you...)

image

you can then do `git reset --hard origin/wmbindata` and your local working copy should be back in sync with the remote copy on github.

**Comment by wimalopaan** [comment]
Thanks, looks good now.

File: radio/src/altdata.h [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        const value_t& operator[](const uint8_t item) const {
+            assert(item < size());
+            return mData[item];
+        }
+        template<typename F>
+        uint8_t pushNextChunk(const uint8_t numberOfValuesToPush, const F f) {
+            const uint8_t start = mNextIndex;
+            for(uint8_t i = 0; i < numberOfValuesToPush; ++i) {
+                const uint8_t v = mData[mNextIndex];
+                TRACE("pushNextData: %d = %d", mNextIndex, v);
```
Comment (3djc) [inline radio/src/altdata.h:34]: Is this TRACE really needed ?

**Pull request closed without merging**